detector/vuls2/vuls2.go

```diff
@@ -221,375 +217,654 @@ func detect(dbc db.DB, sr scanTypes.ScanResult) (detectTypes.DetectResult, error
+						Tag:       tag.tag,
+					},
+				}
+				base := m[src]
+				if !slices.Contains(base.detectableCveIDs, cveID) {
+					base.detectableCveIDs = append(base.detectableCveIDs, cveID)
 				}
+				m[src] = base
 			}
+		}
+	}
 
-			smBase := make(map[string]models.PackageFixStatus)
-			for _, sBase := range sviBase.vulnInfo.AffectedPackages {
-				smBase[sBase.Name] = sBase
+	vim := make(models.VulnInfos)
+	for _, vd := range m {
+		for _, vi := range vd.vulninfos {
+			if !slices.Contains(vd.detectableCveIDs, vi.CveID) {
+				continue
 			}
-			for _, pack := range packs {
-				sBase, found := smBase[pack.name]
-				if found {
-					pack.status = resolvePackageByRootID(family, sviBase.rootID, svi.rootID, sBase, pack.status)
+
+			base, ok := vim[vi.CveID]
+			if ok {
+				merged, err := mergeVulnInfo(base, vi)
+				if err != nil {
+					return nil, xerrors.Errorf("Failed to merge vuln info. err: %w", err)
 				}
-				smBase[pack.name] = pack.status
+				base = merged
+			} else {
+				base = vi
 			}
-			sviBase.vulnInfo.AffectedPackages = slices.Collect(maps.Values(smBase))
-			sviBase.vulnInfo.AffectedPackages.Sort()
-
-			vimBase[vid] = sviBase
+			vim[vi.CveID] = base
 		}
 	}
+	for _, vi := range vim {
+		ps := make(models.PackageFixStatuses, 0, len(am[vi.CveID].packm))
+		for _, p := range am[vi.CveID].packm {
+			ps = append(ps, p.packStatus.status)
+		}
+		vi.AffectedPackages = ps
+		vi.CpeURIs = am[vi.CveID].cpes
 
-	res := models.VulnInfos{}
-	for vid, svi := range vimBase {
-		res[string(vid)] = svi.vulnInfo
+		vim[vi.CveID] = vi
 	}
-	return res, nil
-}
 
-func collectPackages(e ecosystemTypes.Ecosystem, family string, dres detectTypes.DetectResult, ospkgs []scanTypes.OSPackage) (map[dataTypes.RootID]map[sourceTypes.SourceID]map[string]pack, error) {
-	pm := make(map[dataTypes.RootID]map[sourceTypes.SourceID]map[string]pack)
-	for _, detected := range dres.Detected {
-		if _, found := pm[detected.ID]; !found {
-			pm[detected.ID] = make(map[sourceTypes.SourceID]map[string]pack)
-		}
-		for _, detections := range detected.Detections {
-			if detections.Ecosystem != e {
-				continue
-			}
+	return vim, nil
+}
 
-			for sourceID, fconds := range detections.Contents {
-				if _, found := pm[detected.ID][sourceID]; !found {
-					pm[detected.ID][sourceID] = make(map[string]pack)
-				}
+func walkVulnerabilityDetections(m map[source]sourceData, scanned scanTypes.ScanResult, vs []detectTypes.VulnerabilityData) error {
+	for _, v := range vs {
+		for _, d := range v.Detections {
+			for sourceID, fconds := range d.Contents {
 				for _, fcond := range fconds {
-					statuses, ignoresAll, err := collectFixStatuses(family, sourceID, fcond.Criteria, ospkgs)
+					statuses, cpes, _, err := walkCriteria(d.Ecosystem, sourceID, fcond.Criteria, scanned)
 					if err != nil {
-						return nil, xerrors.Errorf("Failed to collect fix statuses. err: %w", err)
+						return xerrors.Errorf("Failed to collect fix statuses. err: %w", err)
```

Comment:
```suggestion
						return xerrors.Errorf("Failed to walk criteria. err: %w", err)
```

Because the function collects cpes, too

Reply:
fixed: https://github.com/future-architect/vuls/compare/e00c803f5f28e58ce052de34fbf6a1d1977a08f4..e3a15408d4270ce78b61ed041c8ac67874cc3c64